openedx/core/djangoapps/user_api/accounts/api.py

```diff
@@ -182,6 +184,13 @@ def update_account_settings(requesting_user, update, username=None):
         for serializer in user_serializer, legacy_profile_serializer:
             serializer.save()
 
+        # if any exception is raised for user preference (i.e. account_privacy), the entire transaction for user account
+        # patch is rolled back and the data is not saved
+        if 'account_privacy' in update:
+            update_user_preferences(
```

Comment:
There is no unit test coverage for this line of code.

https://build.testeng.edx.org/job/edx-platform-python-unittests-pr/5738/Diff_Coverage_Report/